scipy/spatial/transform/_rotation.pyx

```diff
@@ -1175,6 +1288,159 @@ cdef class Rotation:
         else:
             return cls(quat, normalize=False, copy=False)
 
+    @cython.embedsignature(True)
+    @classmethod
+    def from_davenport(cls, axes, angles, extrinsic=True, degrees=False):
+        """Initialize from Davenport angles.
+
+        Rotations in 3-D can be represented by a sequence of 3
+        rotations around a sequence of axes.
+
+        The three rotations can either be in a global frame of reference
+        (extrinsic) or in a body centred frame of reference (intrinsic), which
+        is attached to, and moves with, the object under rotation [1]_.
+
+        For Euler angles, we have the following conditions:
+
+            - ``dot(axis1, axis2) = 0``
+            - ``dot(axis2, axis3) = 0``
+            - ``dot(axis1, axis3) = -1``, ``0`` or ``1``
+
+        For Davenport angles, the third relation is relaxed [2]_:
+            - ``dot(axis1, axis3)`` can be any value between -1 and 1.
+
+        Parameters
+        ----------
+        axes : list of (3, 1) arrays
```

Comment:
I think it should be put as `array_like` with shape (3, 3) where each array or item represents an axis.

Conceptually though, it looks like there is no contradiction in using arbitrary number of axes or axes not being orthogonal. After all it's just sequential composition of rotations. 

Do we need to impose these strict requirements here? Note that it's possible to call `from_euler('Z', 90, degrees=True)`, but the maximum number of axes is 3.

All in all this method is not super useful, but for symmetry it probably should be here.

Reply:
I indeed only put it there for symmetry, but you're right about relaxing the input.

Reply:
I think the most reasonable way is to mirror `from_euler` convention (up to 3 axis). I would preserve orthogonality checks to keep everything within a Davenport framework.